cmake-tool/simulate_scripts/simulate.py

```diff
@@ -47,9 +49,7 @@ def parse_args():
 def notice(message):
     # Don't call this without initialising `progname`.
     assert(progname)
-    sys.stderr.write("{}: {}".format(progname, message))
-    sys.stderr.flush()
-
+    print("{}: {}".format(progname, message), file=sys.stderr)
```

Comment:
Please make this change a separate commit.

Reply:
And please don't remove the flush, the output is not necessarily on a terminal and if a script is calling this somewhere you want the error message in the logs a close to the cause as possible.

Reply:
using `printf(...., flush=true)`is also an option.

Reply:
Nothing else is writing to stderr?

Reply:
> In anything else is writing to stderr?

Not sure I get the context of this question. This script could run in a bigger context and thus there might be more that is written to `stderr`. Maybe even within the same python interpreter session. But from my experience with various CI system (and their Web-GUIs) , it is usually a good idea to force flushes to see then output in the "right" place.

Reply:
Yes, CI is exactly where log reordering happens all the time without flushing.